examples/tts/fastpitch_finetune_adapters.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from dataclasses import is_dataclass
+
+import pytorch_lightning as pl
+import torch
+from omegaconf import DictConfig, OmegaConf, open_dict
+
+from nemo.collections.common.callbacks import LogEpochTimeCallback
+from nemo.collections.tts.models import FastPitchModel
+from nemo.collections.tts.modules.submodules import WeightedSpeakerEmbedding
+from nemo.core import adapter_mixins
+from nemo.core.config import hydra_runner
+from nemo.utils import logging
+from nemo.utils.exp_manager import exp_manager
+
+
+def update_model_config_to_support_adapter(config) -> DictConfig:
+    with open_dict(config):
+        enc_adapter_metadata = adapter_mixins.get_registered_adapter(config.input_fft._target_)
+        if enc_adapter_metadata is not None:
```

Comment:
We should think about whether there is a design that will let us use adapters without having to hardcode every layer in every model that needs to be adapted or frozen.

Reply:
As discussed, this is by design. This is how adapters are implemented in ASR and we just used it as reference.
Do you mean we should give users the flexibility to use adapters to specific layers and not to entire modules like the whole of aligner/encoder etc but just few layers in these modules?